src/test/java/com/next/step/step3/service/WinnerElectionTest.java

```diff
@@ -0,0 +1,66 @@
+package com.next.step.step3.service;
+
+import com.next.step.step3.domain.Car;
+import com.next.step.step3.domain.Cars;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class WinnerElectionTest {
+
+    @ParameterizedTest(name = "자동차가 레이싱 완료 후 한 명의 우승자가 존재하는지 테스트")
+    @MethodSource("provideOneWinnerCars")
+    void electionWinners_우승자_한_명_테스트(Cars cars, List<String> expected) {
+        List<String> result = WinnerElection.electionWinners(cars);
+        assertThat(result).containsAnyElementsOf(expected);
+    }
+
+    static Stream<Arguments> provideOneWinnerCars() {
+        return Stream.of(
+                Arguments.of(new Cars(oneWinnerCars()), List.of("c"))
+        );
+    }
+
+    private static List<Car> oneWinnerCars() {
+        return List.of(
+                new Car(1, "a"),
+                new Car(2, "b"),
+                new Car(3, "c")
+        );
+    }
+
+    @ParameterizedTest(name = "자동차가 레이싱 완료 후 여러 명의 우승자가 존재하는지 테스트")
+    @MethodSource("provideManyWinnerCars")
+    void electionWinners_우승자_여러_명_테스트(Cars cars, List<String> expected) {
+        List<String> result = WinnerElection.electionWinners(cars);
+        assertThat(result).containsAnyElementsOf(expected);
+    }
+
+    static Stream<Arguments> provideManyWinnerCars() {
+        return Stream.of(
+                Arguments.of(new Cars(manyWinnerCarsCase1()), List.of("a", "b", "c")),
+                Arguments.of(new Cars(manyWinnerCarsCase2()), List.of("b", "c"))
+        );
+    }
+
+    private static List<Car> manyWinnerCarsCase1() {
+        return List.of(
+                new Car(1, "a"),
+                new Car(1, "b"),
+                new Car(1, "c")
+        );
+    }
+
+    private static List<Car> manyWinnerCarsCase2() {
+        return List.of(
+                new Car(1, "a"),
+                new Car(100, "b"),
+                new Car(100, "c")
+        );
+    }
```

Comment:
```suggestion
    private static Stream<Arguments> provideManyWinnerCars() {
        return Stream.of(
                Arguments.of(모두_우승한_경우_CARS, List.of("a", "b", "c")),
                Arguments.of(B_C가_우승한_경우_CARS, List.of("b", "c"))
        );
    }

    private static final Cars 모두_우승한_경우_CARS = new Cars(List.of(
            new Car(1, "a"),
            new Car(1, "b"),
            new Car(1, "c"))
    );

    private static final Cars B_C가_우승한_경우_CARS = new Cars(List.of(
            new Car(1, "a"),
            new Car(100, "b"),
            new Car(100, "c"))
    );
```

이 부분은 취향차가 있을듯 한데요.
- 재사용하는 테스트 메소드의 경우 한글명이 직관적이라고 생각합니다.
- 테스트 조건을 구성하는 메소드들은 별도의 Fixture 역할을 하는 객체로 도출하는 편입니다.


그리고 위 상황에선 static 메소드를 상수로 표현할 수도 있겠네요.

Reply:
오 꿀팁 감사합니다:) CarsFactoryTest와 CarNamesExtractorTest는 MethodSource가 많아지면 Fixture로 빼는 방향으로 해볼게요 ㅎㅎ